### PR TITLE
Fix-80065 Failed CLI extension installation results in exit code 0 

### LINKED ISSUE #80065: Failed CLI extension installation results in exit code 0
Issue Type: <b>Bug</b>

When an extension fails to install via the command line, the exit code should not indicate success.

```powershell
> echo "" > test.vsix
> codium --install-extension .\test.vsix
Installing extensions...
Corrupt ZIP: end of central directory record signature not found
Failed Installing Extensions: .\test.vsix
> echo $lastexitcode
0
```

0 indicates that the operation was successful, even though it was not.

VS Code version: VSCodium 1.37.1 (9afbbd903b7b481aa127d4d8d68626397e6a5135, 2019-08-16T10:21:32.989Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2904)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.85GB (15.57GB free)|
|Process Argv|--snip--|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (10)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|2.0.5
pascal|ale|8.0.2
pascal-formatter|ale|2.2.0
githistory|don|0.4.6
gitlens|eam|9.9.3
--snip--
innosetup|idl|1.2.0
powershell|ms-|2019.5.0
vscode-typescript-tslint-plugin|ms-|1.2.2
vim|vsc|1.10.0

</details>

### PR DESCRIPTION
**Pull request by skprabhanjan**
@sandy081 , Here is the fix to make the exit code 1 explicitly when the installation fails to fix #80065 .
Please review this and let me know if anything needs to be changed :)

### CODE REVIEW

**Comment by skprabhanjan** [comment]
@sandy081 , Were you able to review this ?

**Comment by skprabhanjan** [comment]
@sandy081 , Any updates on this ?

File: src/vs/code/node/cliProcessMain.ts [hunk 1]
```diff
@@ -147,7 +147,7 @@ export class Main {
 		if (installedExtensionsManifests.some(manifest => isLanguagePackExtension(manifest))) {
 			await this.updateLocalizationsCache();
 		}
-		return failed.length ? Promise.reject(localize('installation failed', "Failed Installing Extensions: {0}", failed.join(', '))) : Promise.resolve();
+		return failed.length ? (Promise.reject(localize('installation failed', "Failed Installing Extensions: {0}", failed.join(', '))), process.exit(1)) : Promise.resolve();
```
Comment (sandy081) [inline src/vs/code/node/cliProcessMain.ts]: Is not the error should be handled at the root (where the main is called)? 

https://github.com/microsoft/vscode/blob/master/src/vs/code/node/cli.ts#L394

And I see the process exits with error code 1 when there are errors. Please check if this code is not being executed and if so why.

**Review by sandy081 — CHANGES_REQUESTED** [review]

File: src/vs/code/node/cliProcessMain.ts (hunk 1 above)
Comment (skprabhanjan) [reply src/vs/code/node/cliProcessMain.ts]: @sandy081 , It is coming into the error part and calling `eventuallyExit` but not sure why the exit code is not set .
In my system when  I check ( `echo $?`) it is correctly set as 1.
Can you please help me here to go forward ?

**Comment by sandy081** [comment]
Don't worry. It works correctly in vs code. Issue is in a different product.

**Pull request closed without merging**